src/ImageKit/ImageKit.php

```diff
@@ -106,8 +106,31 @@ public function __construct($publicKey, $privateKey, $urlEndpoint, $transformati
      * @param array $options
      * @return string
      */
-    public function url(array $options)
+    public function url($options=null)
     {
+        if(!isset($options)){
+            return json_encode(Response::respond(true, ((object)ErrorMessages::$URL_GENERATION_PARAMETER_MISSING)));
+        }
+        if(!is_array($options)){
+            return json_encode(Response::respond(true, ((object)ErrorMessages::$URL_GENERATION_PARAMETER_NON_ARRAY)));
+        }
+        if(sizeof($options)==0){
+            return json_encode(Response::respond(true, ((object)ErrorMessages::$URL_GENERATION_PARAMETER_EMPTY_ARRAY)));
+        }
+        if (isset($options['src']) && !filter_var($options['src'], FILTER_VALIDATE_URL)) {
+            return json_encode(Response::respond(true, ((object)ErrorMessages::$URL_GENERATION_URL_INVALID)));
```

Comment:
The error message should be `src is not a valid URL`

Reply:
done